spring-boot/proxy/src/test/java/ApiKeyAuthTokenVerifierTest.java

```diff
@@ -0,0 +1,185 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+import eu.coatrack.api.ApiKey;
+import eu.coatrack.api.ServiceApi;
+import eu.coatrack.proxy.security.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.security.web.authentication.session.SessionAuthenticationException;
+import org.springframework.web.client.RestClientException;
+
+import java.net.ConnectException;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ApiKeyAuthTokenVerifierTest {
+
+    //Verification via admin key is not tested yet because it is unclear if this feature will outlast.
+
+    private final String someValidApiKeyValue = "ca716b82-745c-4f6d-a38b-ff8fe140ffd1";
+    private ApiKeyAuthToken apiKeyAuthToken;
+    private ApiKeyAuthTokenVerifier apiKeyAuthTokenVerifier;
+    private ApiKey apiKey;
+
+    private LocalApiKeyManager localApiKeyManagerMock;
+    private ApiKeyFetcher apiKeyFetcherMock;
+    private LocalApiKeyVerifier localApiKeyVerifierMock;
+
+    @BeforeEach
+    public void setup(){
+        apiKeyAuthTokenVerifier = createApiKeyAuthTokenVerifier();
+
+        apiKeyAuthToken = new ApiKeyAuthToken(someValidApiKeyValue, null);
+        apiKeyAuthToken.setAuthenticated(false);
+
+        apiKey = createApiKey();
+    }
+
+    private ApiKeyAuthTokenVerifier createApiKeyAuthTokenVerifier() {
+        localApiKeyManagerMock = mock(LocalApiKeyManager.class);
+        apiKeyFetcherMock = mock(ApiKeyFetcher.class);
+        localApiKeyVerifierMock = mock(LocalApiKeyVerifier.class);
+
+        ApiKeyAuthTokenVerifier localApiKeyAuthTokenVerifier = new ApiKeyAuthTokenVerifier(
+                localApiKeyManagerMock,
+                apiKeyFetcherMock,
+                localApiKeyVerifierMock
+        );
+
+        return localApiKeyAuthTokenVerifier;
+    }
+
+    private ApiKey createApiKey() {
+        ServiceApi serviceApi = new ServiceApi();
+        serviceApi.setUriIdentifier("");
+
+        apiKey = new ApiKey();
+        apiKey.setKeyValue(someValidApiKeyValue);
+        apiKey.setServiceApi(serviceApi);
+
+        return apiKey;
+    }
+
+    @Test
+    public void nullArgumentShouldCauseException(){
+        assertThrows(SessionAuthenticationException.class, () -> apiKeyAuthTokenVerifier.authenticate(null));
+    }
+
+    @Test
+    public void nullCredentialsInAuthTokenShouldCauseException(){
+        String nullCredentials = null;
```

Comment:
Do we really need this line? Would probably be easier to understand if we would remove it or replace it by a comment.

Reply:
I deleted this line.